params/config.go

```diff
@@ -341,6 +356,17 @@ func (c *ChainConfig) checkCompatible(newcfg *ChainConfig, headHeight *big.Int,
 		return newCompatError("SubnetEVM fork block timestamp", c.SubnetEVMTimestamp, newcfg.SubnetEVMTimestamp)
 	}
 
+	var allowListTimestamp, newAllowListTimestamp *big.Int
+	if c.AllowListConfig != nil {
+		allowListTimestamp = c.AllowListConfig.Timestamp()
+	}
+	if newcfg.AllowListConfig != nil {
+		newAllowListTimestamp = newcfg.AllowListConfig.Timestamp()
+	}
+	if isForkIncompatible(allowListTimestamp, newAllowListTimestamp, headTimestamp) {
```

Comment:
❤️

Reply:
I think we should instruct everyone to structure their modifications as hard forks so it is easy to discern changes.